OpenRobertaServer/staticResources/js/app/roberta/controller/galleryList.controller.js

```diff
@@ -120,6 +156,41 @@ define([ 'require', 'exports', 'log', 'util', 'comm', 'message', 'guiState.contr
         $('#galleryTable').on('shown.bs.collapse hidden.bs.collapse', function(e) {
             $('#galleryTable').bootstrapTable('resetWidth');
         });
+
+        $('#filterRobot').onWrap('change', loadGalleryData, "gallery filter changed");
+
+        $('#fieldOrderBy').change(function (e) {
+            var fieldData = e.target.value.split(':');
+            var row = parseInt(fieldData[0]);
+            console.log(row);
+            $('#galleryTable').bootstrapTable('refreshOptions', {
+                sortName: row,
+                sortOrder: fieldData[1],
+            });
+        });
+
+        $('#toogleView').click(function (e) {
+            // toggle button icon
+            var iconClassName = '';
+            if (currentViewMode === 'gallery') {
+                currentViewMode = 'list';
+                iconClassName = 'typcn-th-list';
```

Comment:
Once you are in the list view I would expect to see the tile button and vice versa